pkg/metrics/instrumenting.go

```diff
@@ -42,12 +44,7 @@ func SetupInstrumentation(prometheusAddr string, serviceName string) error {
 	provider := metric.NewMeterProvider(metric.WithReader(exporter, v))
 	global.SetMeterProvider(provider)
 
-	registry := prometheus.NewRegistry()
-	if err := registry.Register(exporter.Collector); err != nil {
-		return fmt.Errorf("error registering collector: %v", err)
-	}
-
-	http.Handle("/metrics", promhttp.HandlerFor(registry, promhttp.HandlerOpts{}))
+	http.Handle("/metrics", promhttp.Handler())
```

Comment:
Looks like this got simplified. It felt suspicious to me since now it magically wires that `registry` using some global registry, from where `otelprom.New()` seems to automatically attach to.